### ISSUE TITLE
ncgeovariable.grid_interop() fails to getCoordinateSystem() on some datasets

### ISSUE THREAD

**Issue by drf5n** [nctoolbox/nctoolbox#65]
This might be related to #64 as well:

    uri='http://web.vims.edu/~drf/tmp/ocean_his_008.nc'
    nc = ncgeodataset(uri);
    gv = nc{'salt'};
    g = gv.grid_interop(1,1,:,:);

    Attempt to reference field of non-structure array.

    Error in ncgeovariable/grid_interop (line 221)
                                        grid = grid.getCoordinateSystem();

    Error in ncgeovariable/subsref (line 1051)
                                    sref = obj.grid_interop(first, last, stride);

I'll try testing if the grid exists after ncgeovariable line 220.

**Comment by rsignell-usgs** [comment]
I don't think #64 is a bug because the dataset is not CF-compliant, so I'm not going to merge this yet. 
Okay @drf5n?

**Comment by drf5n** [comment]
The failure might be a bit bug-like with unexpected and inconsistent behavior--If it didn't have the ocean_z coordinate and was a 2d field like 'ubar', grid_interop passes the x_rho, y_rho unchanged and still interop-transforms the time variable:

    uri='http://web.vims.edu/~drf/tmp/ocean_his_008.nc'
    nc = ncgeodataset(uri);
    gv = nc{'ubar'};
    g = gv.grid_interop(1,:,:)

    g = 

     x_u: [72x51 double]
     y_u: [72x51 double]
    time: 718941

    datestr(g.time)

    ans =

    23-May-1968

grid_interop doesn't fault on this similarly non-cf dataset.  If it should instead fail and throw an error, maybe it should also check for success of findGridByName() and then fault out with a more informative error about non-CF compliance.  Or maybe both: Warn about non-compliance and then do its best effort.

**Comment by drf5n** [comment]
The current 1.0.x branch handles this as expected.  See the pretty non-georeferenced plot resulting from geodemo_1e.m:

    cd ~/matlab/nctoolbox/demos
    geodemo_1e

I'm closing this issue.

![geodemo_1e](https://cloud.githubusercontent.com/assets/2236516/6619807/2699b984-c8a3-11e4-91cb-e65bd36ff6fe.png)

**Comment by drf5n** [comment]
closed.

**Comment by rsignell-usgs** [comment]
@drf5n ,nice plot! 
Do you anticipate more improvements in the short term? 
If not, we should tag a release as 1.1.2 or 1.2.0, as you see fit.

**Comment by drf5n** [comment]
This was a bugfix that I noticed in trying to help students apply the demos to some test-case data from a ROMS RIVERPLUME2 testcase Courtney was using in her class.   I don't see anything else to add so far.

Per http://semver.org/ I'd say it was a backwards-compatible bug-fix at the patch level to make a 1.1.2.

How is one expected to install and maintain nctoolbox?    I've got branch 1.0.x checked out with git and am doing pulls, and this change is now merged in there.    It might be nice to add a command that tells folks what branch,  release, and/or version(s) they are using.  It could list the java/ directory as well and maybe help with debugging and troubleshooting as well.
